src/main/scala/io/scalac/tezos/translator/routes/dto/LibraryEntryRoutesDto.scala

```diff
@@ -1,43 +1,43 @@
 package io.scalac.tezos.translator.routes.dto
 
-import io.scalac.tezos.translator.model.LibraryEntry.PendingApproval
+import io.scalac.tezos.translator.model.LibraryEntry.{PendingApproval, Status}
 import io.scalac.tezos.translator.model.{EmailAddress, LibraryEntry, Uid}
 
 import scala.util.{Success, Try}
 
 case class LibraryEntryRoutesDto(
-  name: String,
-  author: String,
+                                  title: String,
+  author: Option[String],
   email: Option[String],
-  description: String,
+  description: Option[String],
   micheline: String,
   michelson: String
 ) {
   def toDomain: Try[LibraryEntry] = {
-    val emailAddress = email match {
-      case Some(v) => EmailAddress.fromString(v).map(Some(_))
-      case None => Success(None)
-    }
-
-    emailAddress.map { ea =>
+    for {
```

Comment:
please use flatMap instead of for comprehesion

Reply:
I wasn't sure how to good use flatMap, so I made something with IMHO looks good. Please correct me or give me little help with that if my solution don't satisfy you.